### PR TITLE
homebrew_cask: Update documentation and examples

### LINKED ISSUE #34696: homebrew_cask: add cask upgrade functionality
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
homebrew_cask

##### SUMMARY
Homebrew supports now cask upgrades via `brew cask upgrade`, but the Ansible module `homebrew_cask` does not. Please add a parameter and functionality to run a cask upgrade.

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

* Remove unavailable `upgrade` module argument.
* Update documentation.
* Update examples.

Fixes: #34696

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE 
- Bugfix Pull Request

##### COMPONENT NAME
changelogs/fragments/34696-homebrew_documentation.yml
lib/ansible/modules/packaging/os/homebrew_cask.py

### CODE REVIEW

**Comment by Akasurde** [comment]
@felixfontein @Andersson007 Could you please review this? Thanks in advance.

File: changelogs/fragments/34696-homebrew_documentation.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- Update documentation and module formatting in homebrew_cask module (https://github.com/ansible/ansible/issues/34696).
```
Comment (felixfontein) [inline changelogs/fragments/34696-homebrew_documentation.yml]: Docs-only changes usually don't get changelog entries. I'm not sure whether this change is important enough to warrant one, but if it does, how about the following instead:
```
bugfixes:
- "homebrew_cask - properly document ``upgrade`` option as an alias to ``upgrade_all``, and not a separate option (https://github.com/ansible/ansible/issues/34696).
```

**Review by felixfontein — APPROVED** [review]
Changes to the module look good!

**Review by felixfontein — APPROVED** [review]
LGTM

**Comment by Akasurde** [comment]
@felixfontein Thanks for the review.

File: lib/ansible/modules/packaging/os/homebrew_cask.py [hunk 2]
```diff
@@ -1,140 +1,152 @@
...
+    type: list
+  path:
+    description:
+    - "':' separated list of paths to search for 'brew' executable."
+    default: '/usr/local/bin'
+    type: path
+  state:
+    description:
+    - State of the cask.
+    choices: [ 'present', 'absent', 'upgraded' ]
```
Comment (Andersson007) [inline lib/ansible/modules/packaging/os/homebrew_cask.py:48]: to be pedantic the following would be better
```
choices: [ 'absent', 'present', 'upgraded' ]
```

File: lib/ansible/modules/packaging/os/homebrew_cask.py [hunk 3]
```diff
@@ -1,140 +1,152 @@
...
-              available
-        type: bool
-        default: 'no'
-        version_added: "2.7.0"
+  name:
+    description:
+    - Name of cask to install or remove.
+    required: true
+    aliases: ['pkg', 'package', 'cask']
+    type: list
```
Comment (Andersson007) [inline lib/ansible/modules/packaging/os/homebrew_cask.py:39]:
```
type: list
elements: str
```

File: lib/ansible/modules/packaging/os/homebrew_cask.py [hunk 4]
```diff
@@ -1,140 +1,152 @@
...
+    description:
+    - State of the cask.
+    choices: [ 'present', 'absent', 'upgraded' ]
+    default: present
+    type: str
+  sudo_password:
+    description:
+    - The sudo password to be passed to SUDO_ASKPASS.
+    required: false
+    version_added: 2.8
```
Comment (Andersson007) [inline lib/ansible/modules/packaging/os/homebrew_cask.py:55]:
```
55 version_added: 2.8
```
i would add quotes here

File: lib/ansible/modules/packaging/os/homebrew_cask.py [hunk 5]
```diff
@@ -1,140 +1,152 @@
...
+    type: bool
+    default: 'no'
+    aliases: ['update-brew']
+    version_added: "2.2"
+  install_options:
+    description:
+    - Options flags to install a package.
+    aliases: ['options']
+    version_added: "2.2"
+    type: list
```
Comment (Andersson007) [inline lib/ansible/modules/packaging/os/homebrew_cask.py:70]: also would be cool to add  ```elements``` here and so on

File: lib/ansible/modules/packaging/os/homebrew_cask.py [hunk 6]
```diff
@@ -1,140 +1,152 @@
...
+  accept_external_apps:
+    description:
+    - Allow external apps.
+    type: bool
+    default: 'no'
+    version_added: "2.5.0"
```
Comment (Andersson007) [inline lib/ansible/modules/packaging/os/homebrew_cask.py:76]:
```
version_added: "2.5.0"
```
we add new features only in major versions, so no need the fix version here and the other several places

**Review by Andersson007 — COMMENTED** [review]
@Akasurde thanks for mentioning and fixing this! @felixfontein thanks for reviewing!
Sorry for the late reply, was a bit busy yesterday.
I put several notes, @Akasurde , please look
and would be cool to fix ```argument_spec``` by adding parameters' types where needed (maybe something else), i noticed at least one place.
Maybe it'll help to exclude any other related rows from ignore.txt.

**Pull request merged**